devito/dimension.py

```diff
@@ -31,10 +30,24 @@ def __str__(self):
     @property
     def symbolic_size(self):
         """The symbolic size of this dimension."""
-        try:
-            return Number(self.size)
-        except TypeError:
-            return self.rtargs[0].as_symbol
+        return self.rtargs[0].as_symbol
+
+
+class FixedDimension(FixedDimensionArgProvider, Dimension):
+
+    is_Fixed = True
+    """This class defines the behaviour of a dimension whose size is fixed
+       at the time of problem definition and can thus be baked into generated
+       code """
```

Comment:
Spacing seems off here: `"""` should be in next line with a blank before `__init__`.

Reply:
Fixed